pkg/controller/builder/builder_secret.go

```diff
@@ -56,6 +56,7 @@ func (builder *SecretBuilder) PutData(key string, value []byte) *SecretBuilder {
 		data = make(map[string][]byte, 1)
 	}
 	data[key] = value
+	builder.get().Data = data
```

Comment:
https://github.com/apecloud/kubeblocks/pull/5807/files#diff-840970406ceefd84c263207dbcb170c27ad44dff37e9217ab5f1305c0a17971dL39 PutStringData function should `builder.get().Data = data`

Reply:
done